os/net/app-layer/snmp/snmp-ber.c

```diff
@@ -150,7 +160,8 @@ snmp_ber_decode_integer(unsigned char *buf, uint32_t *buff_len, uint32_t *num)
 
   buf = snmp_ber_decode_type(buf, buff_len, &type);
 
-  if(type != BER_DATA_TYPE_INTEGER) {
+  if(buf == NULL ||
+     type != BER_DATA_TYPE_INTEGER) {
```

Comment:
You _could_ keep those in the same line, but you don't have to if you cannot be bothered